diff --git a/bindings/java/hyperic_jni/jni-build.xml b/bindings/java/hyperic_jni/jni-build.xml
index d4556c3..6e59f57 100644
--- a/bindings/java/hyperic_jni/jni-build.xml
+++ b/bindings/java/hyperic_jni/jni-build.xml
@@ -324,6 +324,8 @@
         <compilerarg value="-O2" if="jni.optim"/>
         <compilerarg value="-g" if="jni.debug"/>
         <compilerarg value="-Wall"/>
+        <compilerarg value="-std=gnu89"/>
+        <compilerarg value="-I/usr/include/tirpc"/>
         <compilerarg value="-Werror" if="jni.werror"/>
         <compilerarg value="${jni.gccm}" if="jni.gccm"/>
         <defineset>
diff --git a/src/os/linux/linux_sigar.c b/src/os/linux/linux_sigar.c
index 0a44623..592df9f 100644
--- a/src/os/linux/linux_sigar.c
+++ b/src/os/linux/linux_sigar.c
@@ -45,6 +45,10 @@
 #define PROC_PARTITIONS PROC_FS_ROOT "partitions"
 #define PROC_DISKSTATS  PROC_FS_ROOT "diskstats"

+#ifndef HZ
+#define HZ 100
+#endif
+
 /*
  * /proc/self/stat fields:
  * 1 - pid
diff --git a/src/os/linux/sigar_os.h b/src/os/linux/sigar_os.h
index 29a2ba3..7e1ead7 100644
--- a/src/os/linux/sigar_os.h
+++ b/src/os/linux/sigar_os.h
@@ -70,7 +70,7 @@ struct sigar_t {
 };

 #define HAVE_STRERROR_R
-#ifndef __USE_XOPEN2K
+#if !defined(__USE_XOPEN2K) && defined(__GLIBC__)
 /* use gnu version of strerror_r */
 #define HAVE_STRERROR_R_GLIBC
 #endif
